### PR TITLE
[GOBBLIN-881] Add job tag field that can be used to filter job statuses

### PR DESCRIPTION
**Pull request by jack-moseley**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-881

### Description
- [x] Here are some details about my PR, including screenshots (if applicable):

With multi-dataset support in gaas, it is more convenient to have an extra field that can be used to filter when querying the job status (this would be dataset name in most cases).

This PR allows jobs to include `job.tag` field then when querying jobStatus, specifying tag in the query will return only job statuses with that tag.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Updated test

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by jack-moseley** [comment]
@arjun4084346 please review

**Comment by arjun4084346** [comment]
+1 LGTM.

**Comment by jack-moseley** [comment]
@sv2000 please merge

File: gobblin-runtime/src/main/java/org/apache/gobblin/service/monitoring/FlowStatusGenerator.java [hunk 1]
```diff
@@ -45,14 +46,14 @@
    * @param count
    * @return the latest <code>count</code>{@link FlowStatus}es. null is returned if there is no flow execution found.
    */
-  public List<FlowStatus> getLatestFlowStatus(String flowName, String flowGroup, int count) {
+  public List<FlowStatus> getLatestFlowStatus(String flowName, String flowGroup, int count, String tag) {
```
Comment (sv2000) [inline gobblin-runtime/src/main/java/org/apache/gobblin/service/monitoring/FlowStatusGenerator.java]: Can we update the javadoc to include the tag param? Also, maybe good to explain how the tag field can be used.

File: gobblin-runtime/src/main/java/org/apache/gobblin/service/monitoring/FlowStatusGenerator.java [hunk 2]
```diff
@@ -63,13 +64,20 @@
    * @param flowName
    * @param flowGroup
    * @param flowExecutionId
-   * @return the flow status, null is returned if the flow status does not exist
+   * @param tag
```
Comment (sv2000) [inline gobblin-runtime/src/main/java/org/apache/gobblin/service/monitoring/FlowStatusGenerator.java]: Can you add an explanation for the tag field?

File: gobblin-restli/gobblin-flow-config-service/gobblin-flow-config-service-client/src/main/java/org/apache/gobblin/service/FlowStatusClient.java [hunk 3]
```diff
@@ -134,13 +134,13 @@ public FlowStatus getLatestFlowStatus(FlowId flowId)
    * @return a list of {@link FlowStatus}es corresponding to the latest <code>count</code> executions.
    * @throws RemoteInvocationException
    */
-  public List<FlowStatus> getLatestFlowStatus(FlowId flowId, Integer count)
+  public List<FlowStatus> getLatestFlowStatus(FlowId flowId, Integer count, String tag)
```
Comment (sv2000) [inline gobblin-restli/gobblin-flow-config-service/gobblin-flow-config-service-client/src/main/java/org/apache/gobblin/service/FlowStatusClient.java]: update javadoc for this method.

**Pull request closed without merging**